model/aws.py

```diff
@@ -28,5 +28,25 @@ def access_key_id(self):
     def secret_access_key(self):
         return self.raw['secret_access_key']
 
+    def rotation_cfg_reference(self) -> model.base.CfgElementReference:
+        raw = self.raw.get('rotation_cfg_reference')
+        if raw:
+            return dacite.from_dict(
+                data_class=model.base.CfgElementReference,
+                data=raw,
+            )
+
+        return None
+
+    def iam_user_name(self):
+        return self.raw.get('iam_user_name')
+
     def _required_attributes(self):
-        return ['region','access_key_id','secret_access_key']
+        static_attributes = [
+            'region',
+            'access_key_id',
+            'secret_access_key',
+        ]
+        dynamic_attributes = ['iam_user_name'] if self.rotation_cfg_reference() else []
```

Comment:
it would be nice to document (preferably at the getters for rotation-cfg-element + iam_user_name) what those are intended for.

also, is it actually true that we always need a IAM user for credentials-rotation?

Reply:
AWS cross-rotation **always** requires `iam_user_name`, yes.

As for the reasoning:
boto `iam_client` either implicitly uses its own iam-user (== self-rotation), or requires to explicitly give an user-name (== cross-rotation).
Btw, this holds also true for GCP rotation, however for GCP we [construct the user-name with ServiceAccount E-Mail and project-id](https://github.com/gardener/cc-utils/blob/3311905313346232547b52c8abebd0e24037b706/ccc/gcp.py#L72C5-L72C5).